test/board_test.exs

```diff
@@ -106,4 +128,21 @@ defmodule BoardTest do
             assert Board.valid_move?(board_3, 7) == false
         end
     end
+
+    describe "full?" do
+        test "returns true when board does not have any empty marks" do
+            board = Board.empty_board
+            |> Board.update(0, "X")
+            |> Board.update(1, "O")
+            |> Board.update(2, "X")
+            |> Board.update(3, "X")
+            |> Board.update(4, "X")
+            |> Board.update(5, "O")
+            |> Board.update(6, "O")
+            |> Board.update(7, "X")
+            |> Board.update(8, "O")
```

Comment:
I think you could probably use `reduce` to write this in a less repetitive way.

Reply:
I'll delete this function. I'm no longer using it.